l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -176,39 +437,107 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
     /// the funds would be lost.
     /// @return l2TxHash The L2 transaction hash of deposit finalization
     function deposit(
+        uint256 _chainId,
         address _l2Receiver,
         address _l1Token,
+        uint256 _mintValue,
```

Comment:
Missing the @param documentation 

Also why is it needed? can't you take it from msg.value?

Reply:
I guess it was added in order to comply to an interface? (like in case of non-eth base token hyperchain, this need to be specified)

But in my opinion I think it would be better to just completely separate the workflow deposit base token and deposit other erc20 in order to simplify the design of the bridge.